jsdecode/decode_json.go

```diff
@@ -0,0 +1,96 @@
+package jsdecode
+
+import (
+	"encoding/json"
+	resources "github.com/googleinterns/terraform-cost-estimation/resources"
+	tfjson "github.com/hashicorp/terraform-json"
+	"io/ioutil"
+	"os"
+)
+
+// ComputeIstanceType is the type of ResourceChange and Resource supported
+// by this package, we consider ComputeInstances only for now.
+const ComputeIstanceType = "google_compute_instance"
+
+// TODO maybe add mem/core info
+// ResourceInfo contains the information about Resource, this struct is
+// used to cast interfaces with before/after states of the certain Resource.
+type ResourceInfo struct {
+	ID          string `json:"id,omitempty"`
+	InstanceID  string `json:"instance_id"`
+	Name        string `json:"name,omitempty"`
+	MachineType string `json:"machine_type,omitempty"`
+	Zone        string `json:"zone,omitempty"`
+}
+
+// ResourceStates contains information about the states of resource
+// of type ComputeInstance in plan configuration.
+type ResourceStates struct {
+	Before *resources.ComputeInstance
+	After  *resources.ComputeInstance
+}
+
+// ExtractPlanStruct extracts tfjson.Plan struct from file in provided path
+// and returns the pointer on it if it is possible, otherwise returns error.
+func ExtractPlanStruct(filePath string) (*tfjson.Plan, error) {
```

Comment:
For some future follow-up change: use io.Reader to avoid file operations in this function, so it's easier to unit test (without requiring an actual file).